### PR TITLE
Specify ruby version using .ruby-version

### PR DESCRIPTION
**Pull request by TobiasBales**
This should fix running tests in CI

This also runs CI on macOS since webview does not compile on the default ubuntu images without digging deeper (not sure if it works at all right now)

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ on:
 jobs:
   tests:
     name: Tests
-    runs-on: ubuntu-latest
+    runs-on: macos-latest
```
Comment (CalvinWalzel) [inline .github/workflows/ci.yml:14]: Do we _need_ macos runs? They're 10 times more expensive:
image

Comment (TobiasBales) [reply .github/workflows/ci.yml:14]: But the repo is public so we don't have to pay/have limits afaik

Comment (TobiasBales) [reply .github/workflows/ci.yml:14]: And yes, we seem to do since otherwise compiling webview fails 😢

Comment (TobiasBales) [reply .github/workflows/ci.yml:14]: > GitHub Actions usage is free for standard GitHub-hosted runners in public repositories, and for self-hosted runners.
from [the docs](https://docs.github.com/en/actions/learn-github-actions/usage-limits-billing-and-administration)

Comment (CalvinWalzel) [reply .github/workflows/ci.yml:14]: Thanks for explaining! 👍

**Review by CalvinWalzel — APPROVED** [review]
LGTM

**Pull request merged**